Deprecate the sulu.context kernel parameter (#8055)
sulu/sulu: UPGRADE-2.x.md

# Upgrade

## 2.6.11

### Deprecate usage of `sulu.context` parameter

> The `sulu.context` parameter is deprecated and services should not longer depend on it.

Reviewer: ```suggestion
The `sulu.context` parameter is deprecated and services should no longer depend on it.
```

> If some services behave differently between an `admin` or a `website` request you should depend on

Reviewer: ```suggestion
If some services behave differently between an `admin` or a `website` request, you should depend on
```

[Symfony Firewall Config](https://symfony.com/doc/7.4/security.html#fetching-the-firewall-configuration-for-a-request)
instead.

```php
$suluContext = $this->security?->getFirewallConfig($request)?->getName() === 'admin' ? 'admin' : 'website';
```

> Still best practices is that your services do not depend on such states, instead of controllers or commands
> should be based on the request or command arguments give other parameters to your service to control its behavior.

Reviewer: ```suggestion
Still, best practices is that your services do not depend on such states. Instead controllers or commands
should be based on the request or command arguments, giving other parameters to your service to control its behavior.
```
